{"uid":"e6453d90d6724b82","name":"Exception test","fullName":"tests.forgot_password_test.TestForgotPassword#test_expected_exception_on_page_title","historyId":"288cd0030c92ba34b4ff4b58ed8ec476","time":{"start":1714180399122,"stop":1714180409479,"duration":10357},"description":"Exception catching","descriptionHtml":"<p>Exception catching</p>\n","status":"broken","statusMessage":"selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x560723791e33 <unknown>\n#1 0x560723481567 <unknown>\n#2 0x5607234cc27d <unknown>\n#3 0x5607234cc331 <unknown>\n#4 0x56072350f624 <unknown>\n#5 0x5607234ee38d <unknown>\n#6 0x56072350cae2 <unknown>\n#7 0x5607234ee103 <unknown>\n#8 0x5607234bf1ba <unknown>\n#9 0x5607234bfb8e <unknown>\n#10 0x56072375659b <unknown>\n#11 0x56072375a34b <unknown>\n#12 0x560723742681 <unknown>\n#13 0x56072375aeb2 <unknown>\n#14 0x5607237276af <unknown>\n#15 0x560723781428 <unknown>\n#16 0x560723781600 <unknown>\n#17 0x560723790f84 <unknown>\n#18 0x7fcc3c494ac3 <unknown>","statusTrace":"self = <tests.forgot_password_test.TestForgotPassword object at 0x7f56a1312e70>\n\n    @allure.description(\"Exception catching\")\n    @allure.title(\"Exception test\")\n    @allure.link(\"github.com/allure-examples/\", name=\"Allure Examples\")\n    @allure.issue(\"github.com/allure-examples/allure-examples/issues/1\", name=\"ISSUE-1\")\n    @allure.testcase(\n        \"github.com/allure-examples/allure-examples/issues/2\", name=\"TESTCASE-2\"\n    )\n    def test_expected_exception_on_page_title(self):\n>       self.about_page.click_login_link()\n\ntests/forgot_password_test.py:53: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \npages/about_page.py:20: in click_login_link\n    self.click(self.LOGIN_LINK)\npages/base_page.py:60: in click\n    el: WebElement = self.wait.until(\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"72b9121e389c2fe9d4c9b941800c5ef3\")>\nmethod = <function element_to_be_clickable.<locals>._predicate at 0x7f56a105fba0>\nmessage = ''\n\n    def until(self, method: Callable[[D], Union[Literal[False], T]], message: str = \"\") -> T:\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value does not evaluate to ``False``.\n    \n        :param method: callable(WebDriver)\n        :param message: optional message for :exc:`TimeoutException`\n        :returns: the result of the last call to `method`\n        :raises: :exc:`selenium.common.exceptions.TimeoutException` if timeout occurs\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            time.sleep(self._poll)\n            if time.monotonic() > end_time:\n                break\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: \nE       Stacktrace:\nE       #0 0x560723791e33 <unknown>\nE       #1 0x560723481567 <unknown>\nE       #2 0x5607234cc27d <unknown>\nE       #3 0x5607234cc331 <unknown>\nE       #4 0x56072350f624 <unknown>\nE       #5 0x5607234ee38d <unknown>\nE       #6 0x56072350cae2 <unknown>\nE       #7 0x5607234ee103 <unknown>\nE       #8 0x5607234bf1ba <unknown>\nE       #9 0x5607234bfb8e <unknown>\nE       #10 0x56072375659b <unknown>\nE       #11 0x56072375a34b <unknown>\nE       #12 0x560723742681 <unknown>\nE       #13 0x56072375aeb2 <unknown>\nE       #14 0x5607237276af <unknown>\nE       #15 0x560723781428 <unknown>\nE       #16 0x560723781600 <unknown>\nE       #17 0x560723790f84 <unknown>\nE       #18 0x7fcc3c494ac3 <unknown>\n\n.venv/lib/python3.12/site-packages/selenium/webdriver/support/wait.py:105: TimeoutException","flaky":false,"newFailed":false,"newBroken":false,"newPassed":false,"retriesCount":0,"retriesStatusChange":false,"beforeStages":[{"name":"session_request","time":{"start":1714180199605,"stop":1714180199605,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"_verify_url","time":{"start":1714180199605,"stop":1714180199605,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"base_url","time":{"start":1714180199605,"stop":1714180199605,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"testStage":{"description":"Exception catching","status":"broken","statusMessage":"selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x560723791e33 <unknown>\n#1 0x560723481567 <unknown>\n#2 0x5607234cc27d <unknown>\n#3 0x5607234cc331 <unknown>\n#4 0x56072350f624 <unknown>\n#5 0x5607234ee38d <unknown>\n#6 0x56072350cae2 <unknown>\n#7 0x5607234ee103 <unknown>\n#8 0x5607234bf1ba <unknown>\n#9 0x5607234bfb8e <unknown>\n#10 0x56072375659b <unknown>\n#11 0x56072375a34b <unknown>\n#12 0x560723742681 <unknown>\n#13 0x56072375aeb2 <unknown>\n#14 0x5607237276af <unknown>\n#15 0x560723781428 <unknown>\n#16 0x560723781600 <unknown>\n#17 0x560723790f84 <unknown>\n#18 0x7fcc3c494ac3 <unknown>","statusTrace":"self = <tests.forgot_password_test.TestForgotPassword object at 0x7f56a1312e70>\n\n    @allure.description(\"Exception catching\")\n    @allure.title(\"Exception test\")\n    @allure.link(\"github.com/allure-examples/\", name=\"Allure Examples\")\n    @allure.issue(\"github.com/allure-examples/allure-examples/issues/1\", name=\"ISSUE-1\")\n    @allure.testcase(\n        \"github.com/allure-examples/allure-examples/issues/2\", name=\"TESTCASE-2\"\n    )\n    def test_expected_exception_on_page_title(self):\n>       self.about_page.click_login_link()\n\ntests/forgot_password_test.py:53: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \npages/about_page.py:20: in click_login_link\n    self.click(self.LOGIN_LINK)\npages/base_page.py:60: in click\n    el: WebElement = self.wait.until(\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"72b9121e389c2fe9d4c9b941800c5ef3\")>\nmethod = <function element_to_be_clickable.<locals>._predicate at 0x7f56a105fba0>\nmessage = ''\n\n    def until(self, method: Callable[[D], Union[Literal[False], T]], message: str = \"\") -> T:\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value does not evaluate to ``False``.\n    \n        :param method: callable(WebDriver)\n        :param message: optional message for :exc:`TimeoutException`\n        :returns: the result of the last call to `method`\n        :raises: :exc:`selenium.common.exceptions.TimeoutException` if timeout occurs\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            time.sleep(self._poll)\n            if time.monotonic() > end_time:\n                break\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: \nE       Stacktrace:\nE       #0 0x560723791e33 <unknown>\nE       #1 0x560723481567 <unknown>\nE       #2 0x5607234cc27d <unknown>\nE       #3 0x5607234cc331 <unknown>\nE       #4 0x56072350f624 <unknown>\nE       #5 0x5607234ee38d <unknown>\nE       #6 0x56072350cae2 <unknown>\nE       #7 0x5607234ee103 <unknown>\nE       #8 0x5607234bf1ba <unknown>\nE       #9 0x5607234bfb8e <unknown>\nE       #10 0x56072375659b <unknown>\nE       #11 0x56072375a34b <unknown>\nE       #12 0x560723742681 <unknown>\nE       #13 0x56072375aeb2 <unknown>\nE       #14 0x5607237276af <unknown>\nE       #15 0x560723781428 <unknown>\nE       #16 0x560723781600 <unknown>\nE       #17 0x560723790f84 <unknown>\nE       #18 0x7fcc3c494ac3 <unknown>\n\n.venv/lib/python3.12/site-packages/selenium/webdriver/support/wait.py:105: TimeoutException","steps":[{"name":"Click Login link","time":{"start":1714180399123,"stop":1714180409478,"duration":10355},"status":"broken","statusMessage":"selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x560723791e33 <unknown>\n#1 0x560723481567 <unknown>\n#2 0x5607234cc27d <unknown>\n#3 0x5607234cc331 <unknown>\n#4 0x56072350f624 <unknown>\n#5 0x5607234ee38d <unknown>\n#6 0x56072350cae2 <unknown>\n#7 0x5607234ee103 <unknown>\n#8 0x5607234bf1ba <unknown>\n#9 0x5607234bfb8e <unknown>\n#10 0x56072375659b <unknown>\n#11 0x56072375a34b <unknown>\n#12 0x560723742681 <unknown>\n#13 0x56072375aeb2 <unknown>\n#14 0x5607237276af <unknown>\n#15 0x560723781428 <unknown>\n#16 0x560723781600 <unknown>\n#17 0x560723790f84 <unknown>\n#18 0x7fcc3c494ac3 <unknown>\n\n","statusTrace":"  File \"/home/runner/work/Selenium-Python-Example/Selenium-Python-Example/.venv/lib/python3.12/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/runner/work/Selenium-Python-Example/Selenium-Python-Example/pages/about_page.py\", line 20, in click_login_link\n    self.click(self.LOGIN_LINK)\n  File \"/home/runner/work/Selenium-Python-Example/Selenium-Python-Example/pages/base_page.py\", line 60, in click\n    el: WebElement = self.wait.until(\n                     ^^^^^^^^^^^^^^^^\n  File \"/home/runner/work/Selenium-Python-Example/Selenium-Python-Example/.venv/lib/python3.12/site-packages/selenium/webdriver/support/wait.py\", line 105, in until\n    raise TimeoutException(message, screen, stacktrace)\n","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":true,"stepsCount":0,"attachmentsCount":0,"hasContent":true,"attachmentStep":false},{"name":"public ip address","time":{"start":1714180409493,"stop":1714180409614,"duration":121},"status":"passed","steps":[],"attachments":[{"uid":"5f76dcfdfabf48f5","name":"HTTP logs of http://checkip.amazonaws.com/","source":"5f76dcfdfabf48f5.txt","type":"text/plain","size":359}],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":1,"hasContent":true,"attachmentStep":false}],"attachments":[{"uid":"2c4babf925d8d568","name":"Cookies","source":"2c4babf925d8d568.json","type":"application/json","size":3635},{"uid":"fe73f3dc81452e4","name":"Session Storage","source":"fe73f3dc81452e4.json","type":"application/json","size":358},{"uid":"fbeac7e460ca3fc2","name":"Local Storage","source":"fbeac7e460ca3fc2.json","type":"application/json","size":350},{"uid":"11f7f57f466f1533","name":"Console Logs","source":"11f7f57f466f1533.json","type":"application/json","size":7723},{"uid":"89cec0d35e149ea6","name":"Full Page Screenshot","source":"89cec0d35e149ea6.png","type":"image/png","size":6246649},{"uid":"e43d739bfbb85925","name":"URL","source":"e43d739bfbb85925.uri","type":"text/uri-list","size":23},{"uid":"ea31784df049d548","name":"Network Logs","source":"ea31784df049d548.json","type":"application/json","size":843369}],"parameters":[],"shouldDisplayMessage":true,"stepsCount":2,"attachmentsCount":8,"hasContent":true,"attachmentStep":false},"afterStages":[{"name":"session_request::0","time":{"start":1714180463037,"stop":1714180463037,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"labels":[{"name":"subSuite","value":"Custom sub suite"},{"name":"story","value":"Forgot Password Feature's Functionality"},{"name":"parentSuite","value":"Custom parent suite"},{"name":"epic","value":"Security"},{"name":"suite","value":"Custom suite"},{"name":"severity","value":"critical"},{"name":"tag","value":"security"},{"name":"host","value":"fv-az1020-296"},{"name":"thread","value":"2481-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"tests.forgot_password_test"},{"name":"resultFormat","value":"allure2"}],"parameters":[],"links":[{"name":"TESTCASE-2","url":"https://github.com/allure-examples/allure-examples/issues/2","type":"tms"},{"name":"ISSUE-1","url":"https://github.com/allure-examples/allure-examples/issues/1","type":"issue"},{"name":"Allure Examples","url":"https://github.com/allure-examples/","type":"link"}],"hidden":false,"retry":false,"extra":{"severity":"critical","retries":[],"categories":[{"name":"Test defects","matchedStatuses":[],"flaky":false}],"tags":["security"]},"source":"e6453d90d6724b82.json","parameterValues":[]}